Dockerfile_test

```diff
@@ -0,0 +1,16 @@
+FROM kudobuilder/golang:1.12
+
+WORKDIR $GOPATH/src/github.com/kudobuilder/kudo
+COPY .git-credentials .git-credentials
+
+
+COPY pkg/    pkg/
+COPY config/    config/
+COPY cmd/    cmd/
+COPY go.mod go.mod
+COPY go.sum go.sum
+COPY Makefile Makefile
+
+RUN cp .git-credentials ~/.git-credentials
+
+RUN make test
```

Comment:
I don't think this will be very cache-friendly, depending on Docker version that we use, we can cache the Go path: https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md#run---mounttypecache

If we don't want to depend on a new version of Docker, we would likely want to break it up into sections, to speed up our CI builds and local test runs:

```
COPY Makefile go.mod go.sum .
RUN make download
COPY pkg/ cmd/ .
RUN make test
```

Reply:
good point, thanks :)